consensus/mint-client/src/bin/main.rs

```diff
@@ -166,9 +175,25 @@ fn main() {
             exit(resp.get_result().get_code().value());
         }
 
-        Commands::GenerateMintTx { out, params } => {
+        Commands::GenerateMintTx {
+            out,
+            chain_id,
+            fog_ingest_enclave_css,
+            params,
+        } => {
+            let maybe_fog_bits = fog_ingest_enclave_css.map(|signature| {
+                let env = Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build());
```

Comment:
Most of our existing code-base looks like this right now:

```
$ git grep "EnvBuilder::new()"
admin-http-gateway/src/main.rs:    let env = Arc::new(grpcio::EnvBuilder::new().build());
consensus/mint-client/src/bin/main.rs:            let env = Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build());
consensus/mint-client/src/bin/main.rs:            let env = Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build());
consensus/mint-client/src/bin/main.rs:            let env = Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build());
consensus/mint-client/src/bin/main.rs:            let env = Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build());
consensus/service/src/consensus_service.rs:            EnvBuilder::new()
consensus/service/src/consensus_service.rs:            EnvBuilder::new()
consensus/tool/src/main.rs:    let env = Arc::new(EnvBuilder::new().name_prefix("consensus-tool-grpc").build());
fog/distribution/src/config.rs:                    EnvBuilder::new()
fog/distribution/src/main.rs:        grpcio::EnvBuilder::new()
fog/ingest/client/src/main.rs:    let grpcio_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/ingest/server/src/controller.rs:            grpcio::EnvBuilder::new()
fog/ingest/server/src/server.rs:            grpcio::EnvBuilder::new()
fog/ingest/server/src/server.rs:            grpcio::EnvBuilder::new()
fog/ingest/server/tests/peer_key_sharing.rs:                grpcio::EnvBuilder::new()
fog/ingest/server/tests/tx_recovery.rs:            let grpcio_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/ledger/server/src/server.rs:                grpcio::EnvBuilder::new()
fog/ledger/server/tests/connection.rs:            let grpc_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/ledger/server/tests/connection.rs:            let grpc_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/ledger/server/tests/connection.rs:        let grpc_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/ledger/server/tests/connection.rs:        let grpc_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/load_testing/src/bin/ingest.rs:        let grpcio_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/overseer/server/src/service.rs:        let grpcio_env = Arc::new(grpcio::EnvBuilder::new().build());
fog/report/cli/src/main.rs:    let grpc_env = Arc::new(EnvBuilder::new().name_prefix("cli").build());
fog/report/server/src/server.rs:            grpcio::EnvBuilder::new()
fog/report/server/tests/grpc_apis.rs:    let env = Arc::new(grpcio::EnvBuilder::new().build());
fog/sample-paykit/src/bin/sample_paykit_remote_wallet.rs:        grpcio::EnvBuilder::new()
fog/sample-paykit/src/client_builder.rs:            EnvBuilder::new()
fog/view/load-test/src/main.rs:        EnvBuilder::new()
fog/view/server/src/server.rs:            grpcio::EnvBuilder::new()
fog/view/server/tests/smoke_tests.rs:        let grpcio_env = Arc::new(grpcio::EnvBuilder::new().build());
go-grpc-gateway/testing/src/server.rs:            grpcio::EnvBuilder::new()
ledger/sync/src/test_app/main.rs:        grpcio::EnvBuilder::new()
mobilecoind-dev-faucet/src/lib.rs:        let grpc_env = Arc::new(grpcio::EnvBuilder::new().cq_count(2).build());
mobilecoind-json/src/bin/main.rs:    let env = Arc::new(grpcio::EnvBuilder::new().cq_count(1).build());
mobilecoind/src/config.rs:            grpcio::EnvBuilder::new()
mobilecoind/src/config.rs:            grpcio::EnvBuilder::new()
mobilecoind/src/service.rs:            EnvBuilder::new()
mobilecoind/src/test_utils.rs:        EnvBuilder::new()
util/grpc-admin-tool/src/bin/main.rs:    let env = Arc::new(grpcio::EnvBuilder::new().build());
util/grpc/src/admin_server.rs:                grpcio::EnvBuilder::new()
util/grpc/src/server_cert_reloader.rs:        let env = Arc::new(EnvBuilder::new().build());
util/grpc/src/server_cert_reloader.rs:        let env = Arc::new(EnvBuilder::new().build());
util/grpc/src/server_cert_reloader.rs:        let env = Arc::new(EnvBuilder::new().build());
watcher/src/bin/main.rs:        EnvBuilder::new()
watcher/src/verification_reports_collector.rs:            grpcio::EnvBuilder::new()
```

I don't personally think that a tiny code fragment like `Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build())` is doing enough to justify putting it in a function, there's almost no work happening there. This is what typical code using the idiomatic builder pattern to initialize an object looks like.

Making tiny functions that don't do any work can make code harder to read, because you have to jump around in the editor to see what's going on. If there were something more substantial or some invariant being maintained then it may make sense. But this will not even save us lines of code -- this is a one-liner we're talking about, so we will have more lines of code and not less if we make a function for this.

If you are worried about the duplication of the string literal and possibly mistakes, maybe we can simply use a string constant here instead?

Reply:
I'm think i'm just going to make it so we always initialize the `grpcio::Environment` even if we don't need to make a grpc connection to handle a particular command, it seems that there's no downside to that and it avoids code duplication. 

thanks for bringing this up